charts/tsa/templates/tsa-deployment.yaml

```diff
@@ -0,0 +1,89 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "tsa.fullname" . }}
+{{ include "tsa.namespace" . | indent 2 }}
+  labels:
+    {{- include "tsa.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.server.replicaCount }}
+  selector:
+    matchLabels:
+      {{- include "tsa.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/tsa-configmap.yaml") . | sha256sum }}
+        {{- if .Values.server.podAnnotations }}
+        {{- toYaml .Values.server.podAnnotations | nindent 8 }}
+        {{- end }}
+      labels:
+        {{- include "tsa.selectorLabels" . | nindent 8 }}
+    spec:
+      serviceAccountName: {{ template "tsa.serviceAccountName" . }}
+      # This doesn't actually use Kubernetes credentials, so don't mount them in.
+      automountServiceAccountToken: {{ .Values.server.serviceAccount.mountToken }}
+      containers:
+        - name: {{ template "tsa.fullname" . }}
+          image: "{{ template "tsa.image" .Values.server.image }}"
+          imagePullPolicy: "{{ .Values.server.image.pullPolicy }}"
+          ports:
+{{- include "tsa.containerPorts" .Values.server.service.ports | indent 12 }}
+          args:
+            - "serve"
+            - "--port={{ .Values.server.args.port }}"
+            - "--timestamp-signer={{ .Values.server.args.signer }}"
+            {{- if eq .Values.server.args.signer "tink" }}
+            - "--tink-key-resource={{ .Values.server.args.tink_key_resource }}"
```

Comment:
consider using `fail` https://helm.sh/docs/chart_template_guide/function_list/#fail
to ensure when using a given signer, the required fields are populated rather than starting up passing and empty/null string.

Reply:
I changed the code to use `fail`.